From 3a2f4438ee96a185524220e50b10d0f2a9a1f353 Mon Sep 17 00:00:00 2001
From: Javier Lopez-Gomez <javier.lopez.gomez@cern.ch>
Date: Mon, 1 Feb 2021 22:45:48 +0100
Subject: [PATCH 095/107] Fixes Issue #6812.

In some scenarios, initialization of variables declared by AutoSynthesizer does
not work as expected (more information on the GH issue), e.g.
```
root [2] unsigned int c
(unsigned int) 0
root [3] d=c
Error in <TRint::HandleTermInput()>: std::runtime_error caught: >>> Interpreter
expected relocatable expression
```

This is because the generated VarDecl initializer is missing a LValueToRValue
conversion, where required by the C++ standard.
---
 lib/Sema/SemaExpr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/Sema/SemaExpr.cpp b/lib/Sema/SemaExpr.cpp
index 87775fd4be..357a5dafee 100644
--- a/lib/Sema/SemaExpr.cpp
+++ b/lib/Sema/SemaExpr.cpp
@@ -12540,7 +12540,7 @@ ExprResult Sema::CreateBuiltinBinOp(SourceLocation OpLoc,
             }
             VD->setTypeSourceInfo(C.getTrivialTypeSourceInfo(ResTy));
             VD->setType(ResTy);
-            VD->setInit(RHSExpr);
+            VD->setInit(DefaultLvalueConversion(RHSExpr).get());
             PushOnScopeChains(VD, getCurScope(), /*Add to ctx*/true);
 
 
-- 
2.35.1

